frontend/src/app/core/global_search/input/global-search-input.component.ts

```diff
@@ -217,8 +223,13 @@ export class GlobalSearchInputComponent implements AfterViewInit, OnDestroy {
     }
   }
 
+  public onClose() {
+    this.ngSelectComponent.ngSelectInstance.searchTerm = this.currentValue;
+  }
+
   public clearSearch() {
-    this.currentValue = this.ngSelectComponent.ngSelectInstance.searchTerm = '';
+    this.currentValue = '';
+    this.ngSelectComponent.ngSelectInstance.searchTerm = '';
```

Comment:
This part (`this.ngSelectComponent.ngSelectInstance.searchTerm`) is called quite often in the file. Please extract it to a method, ideally a setter.